sphinx_needs/builder.py

```diff
@@ -157,3 +155,74 @@ def build_needumls_pumls(app: Sphinx, _exception: Exception) -> None:
         needs_builder.set_environment(env)
 
     needs_builder.finish()
+
+
+class NeedsIdBuilder(Builder):
+
+    """Json builder for needs, which creates separate json-files per need"""
+
+    name = "needs_id"
+    format = "needs"
+    file_suffix = ".txt"
+    links_suffix = None
+
+    def write_doc(self, docname: str, doctree: nodes.document) -> None:
+        pass
+
+    def finish(self) -> None:
+        env = unwrap(self.env)
+        needs = env.needs_all_needs.values()
+        filters = env.needs_all_filters
+        config = env.config
+        version = getattr(config, "version", "unset")
+
+        filter_string = self.app.config.needs_builder_filter
+        filtered_needs = filter_needs(self.app, needs, filter_string)
+        needs_build_json_per_id_path = self.app.config.needs_build_json_per_id_path
+        for need in filtered_needs:
+            needs_list = NeedsList(config, self.outdir, self.srcdir)
+            needs_list.wipe_version(version)
+            needs_list.add_need(version, need)
+            for need_filter in filters.values():
```

Comment:
I think adding the filters make no sense is in this case, as they would get repeated in each json file.
Just ignore them.